concurrent/calculator/solution/calculator.erl

```diff
@@ -0,0 +1,48 @@
+-module(calculator).
+-export([start_calculator/0,
+         calculator_server/0,
+         add/3,
+         substract/3,
+         multiply/3,
+         divide/3,
+         turn_off/1]).
+
+start_calculator() ->
+  spawn(calculator, calculator_server, []).
+
+send_operation(Pid, Operation, Numbers) ->
+  Pid ! {self(), Operation, Numbers},
+  receive
+    Result ->
+      Result
+  end.
+
+add(Pid, X, Y) ->
+  send_operation(Pid, add, {X, Y}).
+
+substract(Pid, X, Y) ->
+  send_operation(Pid, substract, {X, Y}).
+
+multiply(Pid, X, Y) ->
+  send_operation(Pid, multiply, {X, Y}).
+
+divide(Pid, X, Y) ->
+  send_operation(Pid, divide, {X, Y}).
+
+calculator_server() ->
+  receive
+    {From, add, {X, Y}} ->
```

Comment:
this is ok but I would probably have a single receive here and defer the actual operation to a helper function with different clauses.

Reply:
You mean to something like this?
```erlang
calculator_server() ->
  receive
    {From, Operation, Numbers} ->
      operation(Operation, Numbers);
    off ->
      exit(shutdown)
  end,
  calculator_server().

operation(add, {X, Y}) ->
  X + Y;
...
```